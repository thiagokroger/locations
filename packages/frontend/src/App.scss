@charset "utf-8";
@import url('https://fonts.googleapis.com/css?family=Nunito:400,700');
@import 'bulma';

html {
  padding: 1rem;
  background-color: #f3f3f3;
}

.card {
  padding: 2px 10px 10px 10px;
  background-color: #dce3e4;
  margin: 6px;
  font-size: 0.7rem;
  text-transform: uppercase;
  letter-spacing: 0.1em;

  .temperature {
    font-size: 1rem;
  }

  .clickable {
    cursor: pointer;
  }

}

.row {
  display: flex;
  justify-content: space-between;
}

.align-right {
  justify-content: flex-end;
}

.column-title {
  padding-left: 6px;
  text-transform: uppercase;
  letter-spacing: 0.1em;
  font-size: 1rem;
}

.weather-container {
  .weather-header {
    text-align: center;
    padding-top: 2rem;
  }

  .weather-forecast {
    padding-top: 2rem;
  }

  .temperature-span {
    font-size: 2rem;
  }

  .weather-description {
    font-size: 0.8rem;
    letter-spacing: 0.1em;
    text-transform: uppercase;
    padding-bottom: 20px;
  }

  .forecast-label {
    font-size: 0.6rem;
    letter-spacing: 0.1em;
    text-transform: uppercase;
  }

  .forecast-value {
    padding-bottom: 15px;
    padding-left: 10px;
    font-size: 0.8rem;
    letter-spacing: 0.1em;
    text-transform: uppercase;
  }

  .forecast-day {
    font-size: 2rem;
  }
}
