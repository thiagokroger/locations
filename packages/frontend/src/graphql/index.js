import gql from 'graphql-tag'

export const GET_CITY = gql`
  query getCity($cityName: String) {
    cityWeather(city: $cityName) {
      cityName
      country
      temperature
      latitude
      longitude
      localtime
      utc
      observationTime
      temperature
      weatherCode
      weatherDescriptions
      windSpeed
      windDegree
      windDirection
      humidity
      feelslike
      forecast {
        date
        sunrise
        sunset
        moonrise
        moonset
        moonPhase
        mintemp
        maxtemp
        avgtemp
      }
    }
  }
`

export const GET_CITIES = gql`
  query getCities($cities: [String]) {
    citiesWeathers(cities: $cities) {
      cityName
      temperature
      utc
      weatherDescriptions
    }
  }
`
