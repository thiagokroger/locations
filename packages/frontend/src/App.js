import React, { useState } from 'react'
import './App.scss'
import { PastWeatherColumn, SearchWeather, Weather } from './container'
import { GET_CITIES, GET_CITY } from './graphql'
import { useLazyQuery, useQuery } from '@apollo/react-hooks'

const App = () => {
  const [zipCode, setZipCode] = useState('')
  const pastSearch = JSON.parse(window.localStorage.getItem('pastSearch')) || []
  const [getCity, { loading, data: searchedPlace }] = useLazyQuery(GET_CITY)
  const { data: pastCities, updateQuery, refetch } = useQuery(GET_CITIES, {
    variables: { cities: pastSearch }
  })

  const addNewPlaceToSearch = (city) => {
    const newPastSearch = [...pastSearch, city]
    window.localStorage.setItem('pastSearch', JSON.stringify(newPastSearch))
    updateQuery(async ({ citiesWeathers }) => {
      const { data } = await refetch({ cities: newPastSearch })
      return {
        citiesWeathers: data.citiesWeathers
      }
    })
  }

  const removeSearchValue = (city) => {
    const past = JSON.parse(window.localStorage.getItem('pastSearch')) || []
    const pastSearch = past.filter(value => value !== city)
    window.localStorage.setItem('pastSearch', JSON.stringify(pastSearch))
    updateQuery(({ citiesWeathers }) => ({
      citiesWeathers: citiesWeathers.filter(({ cityName }) => city !== cityName)
    }))
  }

  const searchPlace = async () => {
    const pastCity = searchedPlace ? searchedPlace.cityWeather.cityName : false
    getCity({ variables: { cityName: zipCode } })
    if (pastCity) {
      await addNewPlaceToSearch(pastCity)
    }
    setZipCode('')
  }

  return (
    <div className="App columns">
      <PastWeatherColumn
        pastSearch={pastCities ? pastCities.citiesWeathers : []}
        removeSearchValue={removeSearchValue}
      />
      <div className="column">
        <SearchWeather
          isLoading={loading}
          zipCode={zipCode}
          setZipCode={(value) => setZipCode(value)}
          searchPlace={searchPlace}
        />
        <Weather placeWeather={searchedPlace ? searchedPlace.cityWeather : null} />
      </div>
    </div>
  )
}

export default App
