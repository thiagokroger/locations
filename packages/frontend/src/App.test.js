import React from 'react'
import App from './App'
import { mount } from 'enzyme'
import { GET_CITIES, GET_CITY } from './graphql'
import { MockedProvider } from '@apollo/react-testing'

describe('Main Page - App', () => {
  test('should render without crashing', () => {
    const mocks = [
      {
        request: { query: GET_CITIES },
        result: {
          data: {
            citiesWeathers: [{
              weatherDescriptions: 'description',
              utc: '0',
              temperature: '10',
              cityName: 'City'
            }]
          },
          loading: false
        }
      },
      {
        request: { query: GET_CITY },
        result: {
          data: {
            cityWeather: null
          },
          updateQuery: () => { },
          refetch: () => { }
        }
      }
    ]

    const component = mount(
      <MockedProvider mocks={mocks}>
        <App />
      </MockedProvider>
    )

    expect(component).toMatchSnapshot()
  })
})
