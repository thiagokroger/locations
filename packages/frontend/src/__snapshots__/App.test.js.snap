// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Page - App should render without crashing 1`] = `
<div
  className="App columns"
>
  <div
    className="column is-one-quarter"
  >
    <span
      className="column-title"
    >
      Last searched places
    </span>
  </div>
  <div
    className="column"
  >
    <div
      className="field has-addons has-addons-centered"
    >
      <div
        className="control has-icons-left"
      >
        <input
          className="input is-rounded"
          disabled={false}
          name="search"
          onChange={[Function]}
          placeholder="Zip code"
          type="text"
          value=""
        />
        <span
          className="icon is-left"
        >
          <i
            className="fas fa-globe"
          />
        </span>
      </div>
      <div
        className="control"
      >
        <button
          className="button is-info "
          onClick={[Function]}
        >
          Search
        </button>
      </div>
    </div>
    <div />
  </div>
</div>
`;
